src/main.rs

```diff
@@ -94,10 +94,40 @@ fn default_filename(filename: Option<String>) -> Result<PathBuf, clap::error::Er
     ))
 }
 
+fn assert_no_uncommitted_changes(path: &PathBuf) -> Result<(), clap::error::Error> {
+    // Extract the filename itself, as well as the directory from `path`.
+    assert!(path.is_file());
+    let filename_without_path = path.file_name().unwrap().to_str().unwrap();
+    let directory = path.parent().unwrap().to_str().unwrap();
```

Comment:
`Command::arg` allows `OsStr`s, so you don't need to convert these to utf-8 anymore:
```suggestion
    let filename_without_path = path.file_name().unwrap();
    let directory = path.parent().unwrap();
```

Reply:
Interesting, thanks I didn't realize that... the constant string unwrapping gets confusing... Will resolve this with a local commit since I'm changing some of the comments in this function anyway.s